python/dgl/nn/pytorch/explain/gnnexplainer.py

```diff
@@ -425,3 +398,395 @@ def explain_graph(self, graph, feat, **kwargs):
+        ...                 graph.update_all(fn.u_mul_e('h', 'w', 'm'), fn.sum('m', 'h'))
+        ...             return graph.ndata['h']
+        >>> # Train the model
+        >>> model = Model(len(features[predict_ntype][0]), data.num_classes, g.etypes)
+        >>> optimizer = torch.optim.Adam(model.parameters())
+        >>> for epoch in range(10):
+        ...     logits = model(g, features)[predict_ntype]
+        ...     loss = F.cross_entropy(logits[train_mask], labels[train_mask])
+        ...     optimizer.zero_grad()
+        ...     loss.backward()
+        ...     optimizer.step()
+        >>> # Explain the prediction for node 10
+        >>> explainer = HeteroGNNExplainer(model, num_hops=1)
+        >>> new_center, sg, feat_mask, edge_mask = explainer.explain_node(predict_ntype, 10, g, features)
+        >>> new_center
+        tensor([0])
+        >>> sg.num_edges()
+        10
+        >>> feat_mask
+        {'SCHEMA': tensor([0.2595, 0.2569, 0.2505, 0.2719, 0.2575, 0.2746, 0.2882, 0.2612, 0.2582,
+                    0.2774]),
+        '_Literal': tensor([0.2729, 0.2802, 0.2796, 0.2739, 0.2619, 0.2903, 0.2771, 0.2515, 0.2879,
+                    0.2734]),
+        'bond': tensor([0.3162, 0.2492, 0.2246, 0.2584, 0.2525, 0.2435, 0.2575, 0.2880, 0.3060,
+                    0.2872]),
+        'd': tensor([0.2628, 0.2904, 0.2877, 0.2416, 0.2869, 0.2561, 0.2542, 0.2945, 0.2757,
+                    0.2786]),
+        'hasStructure': tensor([0.2896, 0.2622, 0.2715, 0.3015, 0.2753, 0.2617, 0.2518, 0.2856, 0.2686,
+                    0.2764])}
+        """
+        self.model.eval()
+        num_nodes = graph.num_nodes()
+        num_edges = graph.num_edges()
+
+        # Extract node-centered k-hop subgraph and
+        # its associated node and edge features.
+        sg, inverse_indices = khop_in_subgraph(graph, {ntype: node_id},self.num_hops)
+        inverse_indices = inverse_indices[ntype]
+        sg_nodes = sg.ndata[NID]
+        sg_edges = sg.edata[EID]
+        sg_feats = {}
+        for node_type in feat.keys():
+            sg_feats[node_type] = feat[node_type][sg_nodes[node_type]]
+        for key, item in kwargs.items():
+            if torch.is_tensor(item) and item.size(0) == num_nodes:
+                item = item[sg_nodes]
+            elif torch.is_tensor(item) and item.size(0) == num_edges:
+                item = item[sg_edges]
+            kwargs[key] = item
+
+        # Get the initial prediction.
+        with torch.no_grad():
+            logits = self.model(sg, sg_feats, **kwargs)[ntype]
+            pred_label = logits.argmax(dim=-1)
+
+        feat_mask, edge_mask = self._init_masks(sg, sg_feats)
+
+        params = [*feat_mask.values(), *edge_mask.values()]
+        optimizer = torch.optim.Adam(params, lr=self.lr)
+        
+        if self.log:
+            pbar = tqdm(total=self.num_epochs)
+            pbar.set_description(f'Explain node {node_id}')
```

Comment:
```suggestion
            pbar.set_description(f'Explain node {node_id} with type {ntype}')
```
? Could you locally try it to verify the success of printing it?

Reply:
Okay, let me verify, and I will commit.